extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -875,14 +878,18 @@ static class PrimitiveTransfer<C extends WritableChunk<Values>, B extends Buffer
             this.chunk = chunk;
             this.buffer = buffer;
             context = columnSource.makeFillContext(targetSize);
-            hasMoreDataToBuffer = false;
         }
 
         @Override
         public boolean hasMoreDataToBuffer() {
             return hasMoreDataToBuffer;
         }
 
+        @Override
```

Comment:
Instead of implementing this for each class, you **could** make it a `default` method on the `TransferObject` 
class. `EncodedTransfer` is the only one that actually needs a custom implementation.

What you have is fine, however, and might be better.  default methods can get you in trouble sometimes and we probably need to think carefully about this function with any new implementation of this class.

Reply:
I wanted to do that, but like you explained, defaults were making me uncomfortable because part of the class needs to be specialized for each type and some parts can be default. So I kept it like this to make it easier to read instead of splitting across base and derived types.